packages/MJCore/src/generic/baseEntity.ts

```diff
@@ -673,27 +679,27 @@ export abstract class BaseEntity {
      * * This method loads a single record from the database. Make sure you first get the correct BaseEntity sub-class for your entity by calling Metadata.GetEntityObject() first. From there, you can
      * call this method to load your records.
      * * NOTE: You should not be calling this method directly from outside of a sub-class in most cases. You will use the auto-generated sub-classes that have overriden versions of this method that blow out the primary keys into individual parameters. This is much easier to program against.
-     * @param PrimaryKeyValues An array of objects that contain the field name and value for the primary key of the record you want to load. For example, if you have a table called "Customers" with a primary key of "ID", you would pass in an array with a single object like this: {FieldName: "ID", Value: 1234}. 
+     * @param CompositeKey Wrapper that holds an array of objects that contain the field name and value for the primary key of the record you want to load. For example, if you have a table called "Customers" with a primary key of "ID", you would pass in an array with a single object like this: {FieldName: "ID", Value: 1234}. 
      * *If you had a composite primary key, you would pass in an array with multiple objects, one for each field in the primary key. You may ONLY pass in the primary key fields, no other fields are allowed.
      * @param EntityRelationshipsToLoad Optional, you can specify the names of the relationships to load up. This is an expensive operation as it loads up an array of the related entity objects for the main record, so use it sparingly.
      * @returns true if success, false otherwise
      */
-    public async InnerLoad(PrimaryKeyValues: PrimaryKeyValue[], EntityRelationshipsToLoad: string[] = null) : Promise<boolean> {
+    public async InnerLoad(CompositeKey: CompositeKey, EntityRelationshipsToLoad: string[] = null) : Promise<boolean> {
         if (BaseEntity.Provider == null) {    
             throw new Error('No provider set');
         }
         else{
             const start = new Date().getTime();
-            this.ValidatePrimaryKeyArray(PrimaryKeyValues);
+            this.ValidatePrimaryKeyArray(CompositeKey.KeyValuePairs);
```

Comment:
@JS-BC  another shortcut here. No shortcuts :) Fix this function to take in the compsoite key. Remmeber, the higher order CK class MIGHT have other info in it one day not just the array of KeyValuePairs.